conformance_tests/test_tiovx/test_ext_supplementary_data.c

```diff
@@ -0,0 +1,1435 @@
+    ERROR_EXPECT_STATUS(tivxSetSupplementaryDataObjDesc(obj_desc[3], supp), VX_SUCCESS, "tivxGetSupplementaryDataObjDesc returns VX_SUCCESS with correct parameters");
+    supp->type_name[0]++;
+    ERROR_EXPECT_STATUS(tivxSetSupplementaryDataObjDesc(obj_desc[3], supp), VX_ERROR_INVALID_TYPE, "tivxGetSupplementaryDataObjDesc returns VX_ERROR_INVALID_TYPE when type does not match");
+    ERROR_EXPECT_STATUS(tivxExtendSupplementaryDataObjDesc(obj_desc[3], supp, &test_data, sizeof(test_data)), VX_ERROR_INVALID_TYPE, "tivxExtendSupplementaryDataObjDesc returns VX_ERROR_INVALID_TYPE when type does not match");
+    supp->type_name[0]--;
+    supp->valid_mem_size = sizeof(test_data) / 2;
+    ERROR_EXPECT_STATUS(tivxExtendSupplementaryDataObjDesc(NULL, supp, &test_data, sizeof(test_data)), VX_ERROR_INVALID_REFERENCE, "tivxExtendSupplementaryDataObjDesc returns VX_ERROR_INVALID_REFERENCE when destination is NULL");
+    ERROR_EXPECT_STATUS(tivxExtendSupplementaryDataObjDesc(obj_desc[3], NULL, NULL, sizeof(test_data)), VX_ERROR_INVALID_PARAMETERS, "tivxExtendSupplementaryDataObjDesc returns VX_ERROR_INVALID_REFERENCE when source and user_data are NULL");
+    ERROR_EXPECT_STATUS(tivxExtendSupplementaryDataObjDesc(obj_desc[3], supp, &test_data, sizeof(test_data) + 1), VX_ERROR_INVALID_VALUE, "tivxExtendSupplementaryDataObjDesc returns VX_ERROR_INVALID_REFERENCE when size is too large");
+    ERROR_EXPECT_STATUS(tivxExtendSupplementaryDataObjDesc(obj_desc[2], supp, &test_data, sizeof(test_data)), VX_FAILURE, "tivxExtendSupplementaryDataObjDesc returns VX_FAILURE when destination has no supplementary data");
+    ERROR_EXPECT_STATUS(tivxExtendSupplementaryDataObjDesc(obj_desc[3], supp, &test_data, 3 * sizeof(test_data) / 4), VX_SUCCESS, "tivxExtendSupplementaryDataObjDesc returns VX_SUCCESS with correct parameters");
+    supp->valid_mem_size = sizeof(test_data);
+    supp = tivxGetSupplementaryDataObjDesc(obj_desc[3], NULL);
+    ERROR_EXPECT_STATUS(supp->valid_mem_size, (vx_uint32)(3U * sizeof(test_data) / 4U), "tivxExtendSupplementaryDataObjDesc sets valid_mem_size correctly");
+    if (VX_SUCCESS == tivxMemBufferMap((void *)(uintptr_t)(supp->mem_ptr.host_ptr ), supp->mem_size, (vx_enum)VX_MEMORY_TYPE_HOST, (vx_enum)VX_READ_ONLY))
+    {
+        tivx_obj_desc_memcpy((void *)&test_data, (void *)(uintptr_t)(supp->mem_ptr.host_ptr), supp->mem_size);
+        tivxMemBufferUnmap((void *)(uintptr_t)(supp->mem_ptr.host_ptr), supp->mem_size, (vx_enum)VX_MEMORY_TYPE_HOST, (vx_enum)VX_READ_ONLY);
+        ERROR_CHECK_VX_SUCCESS(test_data.numbers[0] != 1 ||
+                           test_data.numbers[1] != 2 ||
+                           test_data.numbers[2] != 70, "tivxExtendSupplementaryDataObjDesc copies correct data");
+    }
+    else
+    {
+        ERROR_CHECK_VX_SUCCESS(VX_FAILURE, "ERROR - Could not map the supplementary data!");
+    }
+    return VX_SUCCESS;
+}
+
+void tivxAddTargetKernelGeneral(vx_context context)
```

Comment:
Can you please change the name to "testAddTargetKernelGeneral"? This will help us avoid namespace confusion

Reply:
done